### ISSUE TITLE
uname machine returns "unknown"

### ISSUE THREAD

**Issue by mildred** [julienlevasseur/terraform-provider-uname#7]
On many platforms the "machine" variable contains "unknown". This is because goInfo uses `uname -i` instead of `uname -m` to get the hardware platform which is non portable.

```
$ uname --help
Usage: uname [OPTION]...
Print certain system information.  With no OPTION, same as -s.

  -a, --all                print all information, in the following order,
                             except omit -p and -i if unknown:
  -s, --kernel-name        print the kernel name
  -n, --nodename           print the network node hostname
  -r, --kernel-release     print the kernel release
  -v, --kernel-version     print the kernel version
  -m, --machine            print the machine hardware name
  -p, --processor          print the processor type (non-portable)
  -i, --hardware-platform  print the hardware platform (non-portable)
  -o, --operating-system   print the operating system
      --help        display this help and exit
      --version     output version information and exit

GNU coreutils online help: <https://www.gnu.org/software/coreutils/>
Full documentation <https://www.gnu.org/software/coreutils/uname>
or available locally via: info '(coreutils) uname invocation'

```

**Comment by julienlevasseur** [comment]
Hi @mildred, thank you for your interest in this provider and for your heads up about this limitation from goInfo.

When I started this provider, few years ago, I was expecting at some point that the usage of a non portable option from goInfo might bring challenges at some point. I decided to wait for the challenge to be faced to act and here we go.
Since the goInfo repo had no change about the `-i` option, but you raised the concern via this issue, I finally forked the goInfo repo to make these changes (https://github.com/julienlevasseur/goInfo) and the uname provider now use this forked goInfo.

I recommend you then to use the 0.2.3 version of `uname` that will show the machine attribute.
https://registry.terraform.io/providers/julienlevasseur/uname/latest

Encore merci pour ta contribution, n'hésites pas si tu as d'autres commentaires ou suggestion.
Au plaisir.
